GoogleSignIn/Sources/GIDSignIn.m

```diff
@@ -477,6 +475,45 @@ + (GIDSignIn *)sharedInstance {
 - (id)initPrivate {
   self = [super init];
   if (self) {
+    // Get the app's main bundle.
+    NSBundle *bundle = [NSBundle mainBundle];
+    
+    // Retrieve any config parameters from the main bundle.
+    NSString *clientID;
+    id configClientID = [bundle objectForInfoDictionaryKey:kConfigClientIDKey];
+    if ([configClientID isKindOfClass:[NSString class]]) {
+      clientID = configClientID;
+    }
+    
+    NSString *serverClientID;
+    id configServerClientID = [bundle objectForInfoDictionaryKey:kConfigServerClientIDKey];
+    if ([configServerClientID isKindOfClass:[NSString class]]) {
+      serverClientID = configServerClientID;
+    }
+    
+    NSString *hostedDomain;
+    id configHostedDomain = [bundle objectForInfoDictionaryKey:kConfigHostedDomainKey];
+    if ([configHostedDomain isKindOfClass:[NSString class]]) {
+      hostedDomain = configHostedDomain;
+    }
+
+    NSString *openIDRealm;
+    id configopenIDRealm = [bundle objectForInfoDictionaryKey:kConfigOpenIDRealmKey];
+    if ([configopenIDRealm isKindOfClass:[NSString class]]) {
+      openIDRealm = configopenIDRealm;
+    }
+    
+    // Try to construct an initial configuration.  We need at least a client ID.
+    GIDConfiguration *configuration;
```

Comment:
Is this needed elsewhere, or can we declare/instantiate solely within the `if` below?

Reply:
Yes, this is more complex than it needs to be.  Cleaned up.